lib/resourcebuilder/apps.go

```diff
@@ -72,6 +76,34 @@ func (b *deploymentBuilder) Do(ctx context.Context) error {
 		}
 	}
 
+	// the CVO is the only operator which doesn't have knowledge of its own operand (itself).  If we detect the CVO deployment
+	// we need to replace the KUBERNETES_SERVICE_HOST env var with the internal load balancer to be resilient to kube-apiserver
+	// rollouts that cause the localhost server to become non-responsive for multiple minutes.
+	if deployment.Namespace == "openshift-cluster-version" && deployment.Name == "cluster-version-operator" {
+		infrastructureConfig, err := b.infrastructureGetter.Infrastructures().Get("cluster", metav1.GetOptions{})
+		// not found just means that we don't have proxy configuration, so we should tolerate and fill in empty
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+		lbURL, err := url.Parse(infrastructureConfig.Status.APIServerInternalURL)
+		if err != nil {
+			return err
+		}
+		lbHost, lbPort, err := net.SplitHostPort(lbURL.Host)
+		if err != nil {
```

Comment:
```go
package main

import (
	"fmt"
	"net"
	"net/url"
)

func main() {
	config := ""
	lbURL, err := url.Parse(config)
	fmt.Printf("%v %s\n", lbURL, err)
	lbHost, lbPort, err := net.SplitHostPort(lbURL.Host)
	fmt.Printf("%q %s %s\n", lbHost, lbPort, err)
}
```

[gives][1] `missing port in address`.  You probably need to catch `missing port in address` and use `lbPort := ""` for that case.

[1]: https://play.golang.org/p/BgvPzFudICv

Reply:
> [gives](https://play.golang.org/p/BgvPzFudICv) `missing port in address`. You probably need to catch `missing port in address` and use `lbPort := ""` for that case.

since missing values result in no substitution, I'll add a comment explaining and skip the error

Reply:
f0f00ff9b9b1 has the explicit `missing port in address` catch.  Marking this thread resolved.